WordPress/gutenberg
# docgen: Omit unknown type tag from Markdown format output

Related: #18045
Unblocks: #19520

This pull request seeks to resolve an issue where, if `docgen` encounters a type that it cannot parse, the Markdown formatter will output a `null` literal value for the type. With these changes, it will ~simply not output a type at all~ output text indicating that the type is unknown, in order to avoid misleading the reader into thinking the type is `null` when it is not. 

This should be considered an interim solution to the larger problem of substituting Doctrine (#18045) in order to use more complex TypeScript types (#18838).

**Testing Instructions:**

Verify unit tests pass:

```
npm run test-unit
```

Verify there are no changed documentation after running:

```
npm run docs:build
```
